src/components/tx-flow/flows/EnableSafenet/ReviewEnableSafenet.tsx

```diff
@@ -0,0 +1,58 @@
+import { useContext, useEffect } from 'react'
+import { Typography } from '@mui/material'
+import SignOrExecuteForm from '@/components/tx/SignOrExecuteForm'
+import EthHashInfo from '@/components/common/EthHashInfo'
+import { Errors, logError } from '@/services/exceptions'
+import { createEnableGuardTx, createMultiSendCallOnlyTx } from '@/services/tx/tx-sender'
+import { type EnableSafenetFlowProps } from '.'
+import { SafeTxContext } from '@/components/tx-flow/SafeTxProvider'
+import type { MetaTransactionData, SafeTransaction } from '@safe-global/safe-core-sdk-types'
+import { OperationType } from '@safe-global/safe-core-sdk-types'
+import { ERC20__factory } from '@/types/contracts'
+import { UNLIMITED_APPROVAL_AMOUNT } from '@/utils/tokens'
+
+const ERC20_INTERFACE = ERC20__factory.createInterface()
+
+export const ReviewEnableSafenet = ({ params }: { params: EnableSafenetFlowProps }) => {
+  const { setSafeTx, safeTxError, setSafeTxError } = useContext(SafeTxContext)
+
+  useEffect(() => {
+    async function getTxs(): Promise<SafeTransaction> {
+      if (params.tokensForPresetAllowances.length === 0) {
+        return await createEnableGuardTx(params.guardAddress)
```

Comment:
nitpick: no need to have `await` with return

Reply:
I think it's needed. We want to return the transaction, not a promise that resolves to a transaction.

If we remove the await, the caller must handle all the cases (fulfilment, rejection, etc.), but we want to do this in this function.

Reply:
https://jakearchibald.com/2017/await-vs-return-vs-return-await/

Reply:
> Note: Outside of try/catch blocks, return await is redundant. There's even an [ESLint rule to detect it](https://eslint.org/docs/latest/rules/no-return-await), but it allows it in try/catch.

Hmm, I'm confused now. You don't have a try/catch block for this call. Always thought it was redundant as mentioned.

Reply:
I meant that if we were to catch an error, then the await block would be needed. 

After a second look, I don't think there are errors we'd want to catch in this function, so I'll remove the await in the next PR

Btw, from the ESLint rule link, TIL:
> This rule was deprecated in ESLint v8.46.0 with no replacement. The original intent of this rule was to discourage the use of return await, to avoid an extra microtask. However, due to the fact that JavaScript now handles native Promises differently, there is no longer an extra microtask. More technical information can be found in [this V8 blog entry](https://v8.dev/blog/fast-async).